Docs: Add Vite integration guides. (#13397)
ckeditor/ckeditor5: docs/installation/frameworks/vuejs-v2.md

Integrating the rich text editor from source allows you to use the full power of {@link framework/index CKEditor 5 Framework}.

### Vite

This guide assumes that you use `create-vue` as your boilerplate. To get started with Vite and Vue, run the command below.

```bash
npm init vue@latest ckeditor5-vue-example
```

> This command will install and execute `create-vue`, the official project scaffolding tool for Vue. It will also allow you to customize your project, for example, by adding Typescript. Choose your preferred options.

Reviewer: `create-vue` is creating Vite + Vue apps that are using Vue 3. This guide is about Vue 2 component. To use Vue 2.x with Vite one have to configure it manually and use https://github.com/vitejs/vite-plugin-vue2 plugin.

#### Installing necessary packages

> You need two packages to use CKEditor from source with Vue and Vite: the official Vue component and the Vite plugin.

Reviewer: Here we mention two necessary packages, but the below npm command is installing 3 packages

<info-box>
	Using the Vite plugin to build CKEditor 5 from the source in Vite is still in the experimental phase. We encourage you to test it and give us feedback. To read more about integration with Vite or its limitations, check the {@link installation/advanced/integrating-from-source-vite Integrating from source with Vite} guide.
</info-box>

Install necessary packages alongside the default theme using the following command.

```bash

> npm install --save @ckeditor/vite-plugin-ckeditor5 @ckeditor/ckeditor5-vue2 @ckeditor/ckeditor5-theme-lark

Reviewer: Do we need to include `@ckeditor/ckeditor5-theme-lark` here? I know that this is used in `vite.config.js`, but the same is in `vue.config.js` section, where `theme-lark` is used, but installed in the next section. Removing theme-lark from here will make "Using the editor from source" section universal for both methods.

```

#### Configuring `vite.config.js`

Configuring CKEditor with Vue and Vite is simple. Modify the existing config by importing `ckeditor5` and adding it to the list of plugins.

```js
// vite.config.js

import ckeditor5 from '@ckeditor/vite-plugin-ckeditor5';

export default defineConfig( {

> 	plugins: [

Reviewer: In Vue2 apps that are using Vite, there will be vue() plugin used as well. We can show vite.config.js example with that plugin, as this will be only 2 lines more and it will show that ckeditor5() is added along other vite plugins in this config.

Integrating the rich text editor from source allows you to use the full power of {@link framework/index CKEditor 5 Framework}.

### Vite

> You need to install some packages to use CKEditor from source with Vue 2 and Vite: the official CKEditor Vue 2 component, the CKEditor Vite plugin, and a theme. Also, Vite uses Vue 3 by default, so keep that in mind if you want to start from scratch. This guide assumes you have Vue 2 with Vite already configured.

Reviewer: ```suggestion
This guide assumes you have Vue 2 with Vite already configured. `create-vue` is using Vue 3 by default, so keep that in mind if you want to start from scratch. 

You need to install some packages to use CKEditor from source with Vue 2 and Vite: the official CKEditor Vue 2 component and the CKEditor Vite plugin.`
```

We do not install the theme in the next command. Also, Vite as a bundler is framework agnostic, so it's better to mention `create-vue` instead of stating that Vite is using Vue 3 by default. I've added my suggestion, but feel free to rewrite it if needed.

Author: True, Vite is not exclusive to Vue. I think the paragraph is ok, so I'll commit it, and you can take a look.

```bash
# npm 6.x
npm init vite@latest ckeditor5-vue2-example --template vue

# npm 7+, extra double-dash is needed:
npm init vite@latest ckeditor5-vue2-example -- --template vue
```

This command will install and execute `create-vite`, the official project scaffolding tool for Vite.

#### Installing necessary packages

You need some packages to use CKEditor from source with Vue 2 and Vite: the official CKEditor Vue component, the CKEditor Vite plugin, and a theme. Also, Vite assumes you'll use Vue 3, so you need to fix that.

First, uninstall Vue 3 and `@vitejs/plugin-vue`.

```bash
npm uninstall vue @vitejs/plugin-vue
```

Then, install the Vue 2 package and the Vue 2 plugin for Vite.

```bash
npm install --save vue@2 @vitejs/plugin-vue2
```

<info-box>
	Using the Vite plugin to build CKEditor 5 from the source in Vite is still in the experimental phase. We encourage you to test it and give us feedback. To read more about integration with Vite or its limitations, check the {@link installation/advanced/integrating-from-source-vite Integrating from source with Vite} guide.

> </info-box>

Reviewer: Instead of writing a guide for converting Vue 3 app back to Vue 2, let's assume that the user has Vite + Vue 2 application with `@vitejs/plugin-vue2` already configured. 

In my opinion, it's not expected steps and if someone is starting a new project they should use Vue 3. Other than changing packages and vite config, there are also some breaking changes related to Vue usage in source files and that needs to be also changed. We do not need to cover that.

When writing this part of the guide, let's assume that we are writing it for someone who has Vue 2 migrated from Webpack to Vite, and now they need some information on what additional packages they need to install and how to use them to have the editor built from source in their Vite+Vue2 app.

Finally, it's time to install CKEditor packages. Install them using the following command.

```bash
npm install --save @ckeditor/vite-plugin-ckeditor5 @ckeditor/ckeditor5-vue2
```

#### Configuring `vite.config.js`

> Configuring CKEditor with Vue 2 and Vite is pretty simple. Modify the existing config by importing `ckeditor5` and `@vitejs/plugin-vue2`. Add them to the list of plugins.

Reviewer: If you decide to change the above section, this sentence would also need some changes, as then we can assume that `@vitejs/plugin-vue2` is already imported and used.
